sdk/storage/sample/datalake_getting_started.cpp

```diff
@@ -3,11 +3,80 @@
 
 #include "datalake/datalake.hpp"
 #include "samples_common.hpp"
+#include <iostream>
 
 SAMPLE(DataLakeGettingStarted, DataLakeGettingStarted)
 void DataLakeGettingStarted()
 {
+  // Please note that you can always reference test cases for advanced usages.
+
   using namespace Azure::Storage::Files::DataLake;
-  auto client = ServiceClient::CreateFromConnectionString(GetConnectionString());
-  auto response = client.ListFileSystems();
+
+  std::string fileSystemName = "sample-file-system";
+  std::string directoryName = "sample-directory";
+  std::string fileName = "sample-file";
+
+  // Initializing a ServiceClient that can then initialize the FileSystemClient or list file
+  // systems.
+  auto serviceClient = ServiceClient::CreateFromConnectionString(GetConnectionString());
+
+  // List all file systems.
+  std::string continuation;
+  std::vector<FileSystem> fileSystems;
+  do
+  {
+    auto response = serviceClient.ListFileSystems();
+    if (response->Continuation.HasValue())
+    {
+      continuation = response->Continuation.GetValue();
+    }
+    fileSystems.insert(
+        fileSystems.end(), response->Filesystems.begin(), response->Filesystems.end());
+  } while (!continuation.empty());
+
+  // Create file systems.
+  auto fileSystemClient = serviceClient.GetFileSystemClient(fileSystemName);
+  auto iter = std::find_if(
+      fileSystems.begin(), fileSystems.end(), [&fileSystemName](const FileSystem& fileSystem) {
+        return fileSystem.Name == fileSystemName;
+      });
+  if (iter == fileSystems.end())
+  {
+    fileSystemClient.Create();
+  }
```

Comment:
I feel this is too complicated as a sample. Sample is supposed to be easy and succinct, so that users can grasp the main points in a short time.

Creating a new filesystem and finding duplicate, which is the one we want to tell a beginner first?

Reply:
dito.

Reply:
Sorry I cannot agree.

What's the first thing we want our users to learn about DataLake SDK? I'd say to create a filesystem, and after that maybe create a directory or file.

So you used 9 lines of code here, only 1 line is useful for our users to know how to create a filesystem.

Finding a specific string in a vector of strings, why would somebody want to know how to do that?

Reply:
Because we don't have `CreateIfNotExist` or anywhere to tell our customers that how you can find anything in a listed result.
Again, I think with the newly added comment make the code very discoverable.

Reply:
Still cannot agree, maybe we should escalate to @vinjiang

Reply:
Sure, I don't have problem with that.

Reply:
After discussion, I think it's one vote against two to stay as is, so I changed the sequence.